packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +62,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
+      {choices.map(choice => <option key={choice.value} value={choice.value}>{choice.display}</option>)}
```

Comment:
Looking into this at the moment, my initial understanding of controlled forms was not up to snuff when I first wrote this.